packages/map-popup/src/EndpointPopup.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import styled from "styled-components";
+import { FormattedMessage } from "react-intl";
+import { Popup } from "@opentripplanner/base-map";
+import { Location } from "@opentripplanner/types";
+
+import FocusTrapWrapper from "./FocusTrapWrapper";
+
+type EndpointProps = {
+  clearLocation: () => void;
+  forgetHome: () => void;
+  forgetWork: () => void;
+  location?: Location;
+  isWork: boolean;
+  isHome: boolean;
+  rememberAsHome: () => void;
+  rememberAsWork: () => void;
+  setShowPopup?: (arg: boolean) => void;
+  swapLocation: () => void;
+  type: string;
+  UserLocationIcon: any;
+};
+
+const Button = styled.button`
+  background: none;
+  border: none;
+  color: navy;
+  font-family: inherit;
+  font-size: inherit;
+  line-height: inherit;
+  padding-left: 0.2em;
+  :hover {
+    cursor: pointer;
+    text-decoration: underline;
+  }
+`;
+
+export const EndpointPopupContent = ({
+  clearLocation,
+  location,
+  forgetHome,
+  forgetWork,
+  isWork,
+  isHome,
+  rememberAsHome,
+  rememberAsWork,
+  swapLocation,
+  type,
+  UserLocationIcon
+}: EndpointProps): JSX.Element => {
+  const otherType = type === "from" ? "to" : "from";
+  const icon = isWork ? "briefcase" : isHome ? "home" : "map-marker";
+  return (
+    <>
+      <strong>
+        <UserLocationIcon type={icon} />
```

Comment:
Why is this component passed in? Should we have a default so icons render in the story?